spec/workers/subject_removal_worker_spec.rb

```diff
@@ -3,18 +3,60 @@
 RSpec.describe SubjectRemovalWorker do
   let(:subject_id) { 1 }
   let(:feature_name) { "remove_orphan_subjects" }
-  let(:remover) { instance_double(Subjects::Remover) }
+  let(:panoptes_client) { instance_double(Panoptes::Client) }
+  let(:subject_remover) { described_class.new }
+  let(:remover) { instance_double(Subjects::Remover, panoptes_client: panoptes_client) }
 
-  it 'should call the orphan remover cleanup when enabled' do
-    Flipper.enable(feature_name)
-    expect(Subjects::Remover).to receive(:new).with(subject_id).and_return(remover)
-    expect(remover).to receive(:cleanup)
-    subject.perform(subject_id)
+  describe 'flipper enabled' do
+    let(:project) { create(:project) }
+    let!(:subjects) { create_list(:subject, 2) }
+    let!(:subject_sets) { create_list(:subject_set, 2, project: project) }
+    let!(:subject_set) { create(:subject_set, project: project) }
+    let(:discussions_url) { 'https://talk-staging.zooniverse.org/discussions' }
+    let!(:first_subject) { subjects.first }
+    let!(:second_subject) { subjects.second }
+
+    def stub_discussions_request(subject_id)
+      stub_request(:get, discussions_url)
+        .with(query: { focus_id: subject_id, focus_type: 'Subject' })
+        .to_return(status: 200, body: '[]', headers: {})
+    end
+
+    before do
+      Flipper.enable(feature_name)
+    end
+
+    context 'when running orphan remover cleanup function' do
+      it 'should call the orphan remover cleanup when enabled' do
```

Comment:
just to get part of hound off your back
```suggestion
      it 'calls the orphan remover cleanup when enabled' do
```